features/upgrade/monitoring/monitoring-upgrade.feature

```diff
@@ -0,0 +1,77 @@
+Feature: cluster monitoring related upgrade check
+  # @author hongyli@redhat.com
+  @upgrade-prepare
+  @admin
+  Scenario: upgrade cluster monitoring along with OCP - prepare
+    Given I switch to cluster admin pseudo user
+    When I run the :apply client command with:
+      | f          | <%= BushSlicer::HOME %>/features/upgrade/monitoring/testdata/cm-monitoring-retention.yaml |
+      | overwrite  | true |
+    Then the step should succeed
+
+  # @author hongyli@redhat.com
+  # @case_id OCP-29797
+  @upgrade-check
+  @admin
+  Scenario: upgrade cluster monitoring along with OCP
+    Given I switch to cluster admin pseudo user
+    And I use the "openshift-monitoring" project
+    Given all pods in the project are ready
+    Then Then the step should succeed
+    When I run the :get client command with:
+      | resource      | clusteroperator |
+      | resource_name | monitoring      |
+    And evaluation of `@result[:stdout].split(/\n/)[1].split(/\s+/)[4]` is stored in the :degreaded_status clipboard
+    Then the expression should be true> cb.degreaded_status == "False"
+
+    #check retention time
+    When I run the :get client command with:
+      | resource      | prometheus |
+      | resource_name | k8s        |
+      | o             | yaml       |
+    Then the expression should be true> YAML.load(@result[:stdout])["spec"]["retention"] == "3h"
+
+    # get sa/prometheus-k8s token
+    When evaluation of `secret(service_account('prometheus-k8s').get_secret_names.find {|s| s.match('token')}).token` is stored in the :sa_token clipboard
+
+    # curl -k -H "Authorization: Bearer $token" 'https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=machine_cpu_cores'
+    When I run the :exec admin command with:
+      | n                | openshift-monitoring |
+      | pod              | prometheus-k8s-0     |
+      | c                | prometheus           |
+      | oc_opts_end      |                      |
+      | exec_command     | sh                   |
+      | exec_command_arg | -c                   |
+      | exec_command_arg | curl -k -H "Authorization: Bearer <%= cb.sa_token %>" https://prometheus-k8s.openshift-monitoring.svc:9091/api/v1/query?query=machine_cpu_cores |
```

Comment:
@lihongyan1 is this route exposed externally?  Is so we have a step to do query via an external route.

Reply:
We can't do this, please refer Juzhao's comments below

Reply:
@Peter, please share the way to verify data from external route, thx